File: MyDailyBeat/MyDailyBeat/EVCCommonMethods.m
//
//  EVCCommonMethods.m
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 10/19/14.
//  Copyright (c) 2014 eVerveCorp. All rights reserved.
//

#import "EVCCommonMethods.h"

@implementation EVCCommonMethods

+ (UIImage *)imageWithImage:(UIImage *)image scaledToSize:(CGSize)newSize {
    //UIGraphicsBeginImageContext(newSize);
    // In next line, pass 0.0 to use the current device's pixel scaling factor (and thus account for Retina resolution).
    // Pass 1.0 to force exact pixel size.
    UIGraphicsBeginImageContextWithOptions(newSize, NO, 0.0);
    [image drawInRect:CGRectMake(0, 0, newSize.width, newSize.height)];
    UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    return newImage;
}

+ (UIButton *) barButtonSystemItem:(UIBarButtonSystemItem) item withText:(NSString *) title {
    UIBarButtonItem *tempBarButton = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:item target:nil action:nil];
    UIImage *img = tempBarButton.image;
    UIButton *btn = [[UIButton alloc] init];
    [btn setBackgroundImage:img forState:UIControlStateNormal];
    [btn setTitle:title forState:UIControlStateNormal];
    
    return btn;

}

+ (UIImage*) imageWithColor:(UIColor*)color size:(CGSize)size
{
    UIGraphicsBeginImageContext(size);
    UIBezierPath* rPath = [UIBezierPath bezierPathWithRect:CGRectMake(0., 0., size.width, size.height)];
    [color setFill];
    [rPath fill];
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    return image;
}

+ (double) metersForRadius: (double) miles {
    return miles * 1609.34;
}
+ (double) milesForRadius: (double) meters {
    return meters/1609.34;
}

+ (UIImage *)fixOrientationForImage: (UIImage *) inputImage {
    
    // No-op if the orientation is already correct
    if (inputImage.imageOrientation == UIImageOrientationUp) return inputImage;
    
    // We need to calculate the proper transformation to make the image upright.
    // We do it in 2 steps: Rotate if Left/Right/Down, and then flip if Mirrored.
    CGAffineTransform transform = CGAffineTransformIdentity;
    
    switch (inputImage.imageOrientation) {
        case UIImageOrientationDown:
        case UIImageOrientationDownMirrored:
            transform = CGAffineTransformTranslate(transform, inputImage.size.width, inputImage.size.height);
            transform = CGAffineTransformRotate(transform, M_PI);
            break;
            
        case UIImageOrientationLeft:
        case UIImageOrientationLeftMirrored:
            transform = CGAffineTransformTranslate(transform, inputImage.size.width, 0);
            transform = CGAffineTransformRotate(transform, M_PI_2);
            break;
            
        case UIImageOrientationRight:
        case UIImageOrientationRightMirrored:
            transform = CGAffineTransformTranslate(transform, 0, inputImage.size.height);
            transform = CGAffineTransformRotate(transform, -M_PI_2);
            break;
        case UIImageOrientationUp:
        case UIImageOrientationUpMirrored:
            break;
    }
    
    switch (inputImage.imageOrientation) {
        case UIImageOrientationUpMirrored:
        case UIImageOrientationDownMirrored:
            transform = CGAffineTransformTranslate(transform, inputImage.size.width, 0);
            transform = CGAffineTransformScale(transform, -1, 1);
            break;
            
        case UIImageOrientationLeftMirrored:
        case UIImageOrientationRightMirrored:
            transform = CGAffineTransformTranslate(transform, inputImage.size.height, 0);
            transform = CGAffineTransformScale(transform, -1, 1);
            break;
        case UIImageOrientationUp:
        case UIImageOrientationDown:
        case UIImageOrientationLeft:
        case UIImageOrientationRight:
            break;
    }
    
    // Now we draw the underlying CGImage into a new context, applying the transform
    // calculated above.
    CGContextRef ctx = CGBitmapContextCreate(NULL, inputImage.size.width, inputImage.size.height,
                                             CGImageGetBitsPerComponent(inputImage.CGImage), 0,
                                             CGImageGetColorSpace(inputImage.CGImage),
                                             CGImageGetBitmapInfo(inputImage.CGImage));
    CGContextConcatCTM(ctx, transform);
    switch (inputImage.imageOrientation) {
        case UIImageOrientationLeft:
        case UIImageOrientationLeftMirrored:
        case UIImageOrientationRight:
        case UIImageOrientationRightMirrored:
            // Grr...
            CGContextDrawImage(ctx, CGRectMake(0,0,inputImage.size.height,inputImage.size.width), inputImage.CGImage);
            break;
            
        default:
            CGContextDrawImage(ctx, CGRectMake(0,0,inputImage.size.width,inputImage.size.height), inputImage.CGImage);
            break;
    }
    
    // And now we just create a new UIImage from the drawing context
    CGImageRef cgimg = CGBitmapContextCreateImage(ctx);
    UIImage *img = [UIImage imageWithCGImage:cgimg];
    CGContextRelease(ctx);
    CGImageRelease(cgimg);
    return img;
}

@end

File: MDB_Swift/MyDailyBeat/MyDailyBeat/src/EVCCommonMethods.swift

//
//  EVCCommonMethods.swift
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 10/19/14.
//  Copyright (c) 2014 eVerveCorp. All rights reserved.
//
import Foundation
import UIKit
class EVCCommonMethods: NSObject {
    
    class func image(with image: UIImage, scaledTo newSize: CGSize) -> UIImage {
        //UIGraphicsBeginImageContext(newSize);
        // In next line, pass 0.0 to use the current device's pixel scaling factor (and thus account for Retina resolution).
        // Pass 1.0 to force exact pixel size.
        UIGraphicsBeginImageContextWithOptions(newSize, false, 0.0)
        image.draw(in: CGRect(x: CGFloat(0), y: CGFloat(0), width: CGFloat(newSize.width), height: CGFloat(newSize.height)))
        let newImage: UIImage? = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        return newImage!
    }

    class func barButtonSystemItem(_ item: UIBarButtonSystemItem, withText title: String) -> UIButton {
        let tempBarButton = UIBarButtonItem(barButtonSystemItem: item, target: nil, action: nil)
        let img: UIImage? = tempBarButton.image
        let btn = UIButton()
        btn.setBackgroundImage(img, for: .normal)
        btn.setTitle(title, for: .normal)
        return btn
    }

    class func image(with color: UIColor, size: CGSize) -> UIImage {
        UIGraphicsBeginImageContext(size)
        let rPath = UIBezierPath(rect: CGRect(x: CGFloat(0.0), y: CGFloat(0.0), width: CGFloat(size.width), height: CGFloat(size.height)))
        color.setFill()
        rPath.fill()
        let image: UIImage? = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        return image!
    }

    class func fixOrientation(for inputImage: UIImage) -> UIImage {
        
        // No-op if the orientation is already correct
        guard inputImage.imageOrientation != .up else {
            return inputImage
        }
        
        let renderer = UIGraphicsImageRenderer(size: CGSize(width: Int(inputImage.size.width), height: Int(inputImage.size.height)))
            // We need to calculate the proper transformation to make the image upright.
            // We do it in 2 steps: Rotate if Left/Right/Down, and then flip if Mirrored.
        var transform = CGAffineTransform.identity
        switch inputImage.imageOrientation {
            case .down, .downMirrored:
                transform = transform.translatedBy(x: inputImage.size.width, y: inputImage.size.height)
                transform = transform.rotated(by: .pi)
            case .left, .leftMirrored:
                transform = transform.translatedBy(x: inputImage.size.width, y: 0)
                transform = transform.rotated(by: CGFloat(M_PI_2))
            case .right, .rightMirrored:
                transform = transform.translatedBy(x: 0, y: inputImage.size.height)
                transform = transform.rotated(by: -(CGFloat)(M_PI_2))
            case .up, .upMirrored:
                break
        }

        switch inputImage.imageOrientation {
            case .upMirrored, .downMirrored:
                transform = transform.translatedBy(x: inputImage.size.width, y: 0)
                transform = transform.scaledBy(x: -1, y: 1)
            case .leftMirrored, .rightMirrored:
                transform = transform.translatedBy(x: inputImage.size.height, y: 0)
                transform = transform.scaledBy(x: -1, y: 1)
            case .up, .down, .left, .right:
                break
        }
        
        return renderer.image(actions: { (context) in
            let ctx = context.cgContext
            ctx.concatenate(transform)
            switch inputImage.imageOrientation {
            case .left, .leftMirrored, .right, .rightMirrored:
                // Grr...
                let rect = CGRect(x: 0, y: 0, width: inputImage.size.height, height: inputImage.size.width)
                ctx.draw(inputImage.cgImage!, in: rect, byTiling: false)
            default:
                let rect = CGRect(x: 0, y: 0, width: inputImage.size.width, height: inputImage.size.height)
                ctx.draw(inputImage.cgImage!, in: rect, byTiling: false)
            }
        })
    }

    class func meters(forRadius miles: Double) -> Double {
        return miles * 1609.34
    }

    class func miles(forRadius meters: Double) -> Double {
        return meters / 1609.34
    }
}
